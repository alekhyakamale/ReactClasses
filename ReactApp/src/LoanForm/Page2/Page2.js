import React, { Component } from "react";
import {Form} from 'formik';
import { MyInput, CustomRadioGroup, ExpenseField } from "../../Data/Data";

export default class Page1 extends Component {
  render() {
    return (
      <>
      {formikProps => (<Form>
        <div className="div-color">
          <h2>Your Income</h2>
          <div className="row">
            <MyInput
              type="number"
              label="What is you Salary?"
              span="$"
              name="salary"
            />{' '}Per{' '}
                <CustomRadioGroup
                type="radio"
                name="incomeper"
                class="salary-div"
                value={formikProps.values.incomeper}
                options={["year", "month", "fortnight", "week"]}
              />{" "}
          </div>
          <input className="mx-3 hover-magic"
           type="checkbox" name="aftertax"
             value={formikProps.values.afterTax} />
            <label for="aftertax">After Tax</label>
          <CustomRadioGroup
            type="radio"
            label="Do you receive any government allowances or pensions?"
            name="allowances"
            value={formikProps.values.allowances}
            options={["Yes", "No"]}
          />
          <CustomRadioGroup
            type="radio"
            label="Do you receive any rental income?"
            name="rental"
            value={formikProps.values.rental}
            options={["Yes", "No"]}
          />
          <CustomRadioGroup
            type="radio"
            label="Do you receive any other additional income? "
            name="additional"
            value={formikProps.values.additional}
            options={["Yes", "No"]}
          />
        </div>
        <div className="div-color">
          <h2>Your Expenses</h2>
          <p>
            Please provide an accurate view of your share of living expenses by
            completing the following fields. (If you donâ€™t have the expense
            please enter $0). This will help us understand your circumstances so
            we can assess your application and we may validate the expenses
            stated.
          </p>
          <ExpenseField
            name="utilities"
            label="Utilities"
            selected_period={formikProps.values["utilitiesPer"]}
            handleChange={formikProps.handleChange}
            handleBlur={formikProps.handleBlur}
          />
          <ExpenseField
            name="household"
            label="Household"
            selected_period={formikProps.values["householdPer"]}
            handleChange={formikProps.handleChange}
            handleBlur={formikProps.handleBlur}
          />
          <ExpenseField
            name="tv"
            label="TV & Communications"
            selected_period={formikProps.values["tvPer"]}
            handleChange={formikProps.handleChange}
            handleBlur={formikProps.handleBlur}
          />
          <ExpenseField
            name="other"
            label="Other"
            selected_period={formikProps.values["otherPer"]}
            handleChange={formikProps.handleChange}
            handleBlur={formikProps.handleBlur}
          />

          {/* Diplay sum of all expenses */}
          <div>
            <br />
            <hr />
            <h3>Estimated Total Expenses</h3>
            <p>
              $
              <span className="bigDigits">
                {(formikProps.values.utilities || 0) +
                  (formikProps.values.household || 0) +
                  (formikProps.values.tv || 0) +
                  (formikProps.values.other || 0)}
              </span>{" "}
              /month
            </p>
          </div>
        </div>
        <div className="submitFooter">
          <button type="button" onClick={this.props.prevPage}>
            {"< Prev"}
          </button>
          <button type="submit">Submit</button>
        </div>
  
  </Form>
 )}
 </>);
}
}
