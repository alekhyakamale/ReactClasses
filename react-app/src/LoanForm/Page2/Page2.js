import React, { useContext } from "react";
import { MyInput, CustomRadioGroup, ExpenseField } from "../../Data/Data";
import {FormContext} from '../Context';
 
export default function Page1 ({formikProps}) {

  const {prevPage} = useContext(FormContext);

  return (
     <>
        <div className="div-color">
          <h2>Your Income</h2>
          <div className="row">
            <MyInput
              type="number"
              label="What is you Salary?"
              span="$"
              name="salary"
            />{' '}Per{' '}
                <CustomRadioGroup
                type="radio"
                name="incomeper"
                class="salary-div"
                value={formikProps.values.incomeper}
                options={["year", "month", "fortnight", "week"]}
              />{" "}
          </div>
          <input className="mx-3 hover-magic"
           type="checkbox" name="aftertax"
             value={formikProps.values.afterTax} />
            <label for="aftertax">After Tax</label>
          <CustomRadioGroup
            type="radio"
            label="Do you receive any government allowances or pensions?"
            name="allowances"
            value={formikProps.values.allowances}
            options={["Yes", "No"]}
          />
          <CustomRadioGroup
            type="radio"
            label="Do you receive any rental income?"
            name="rental"
            value={formikProps.values.rental}
            options={["Yes", "No"]}
          />
          <CustomRadioGroup
            type="radio"
            label="Do you receive any other additional income? "
            name="additional"
            value={formikProps.values.additional}
            options={["Yes", "No"]}
          /> 
        </div>
        <div className="div-color">
          <h2>Your Expenses</h2>
          <p>
            Please provide an accurate view of your share of living expenses by
            completing the following fields. (If you donâ€™t have the expense
            please enter $0). This will help us understand your circumstances so
            we can assess your application and we may validate the expenses
            stated.
          </p>
          <ExpenseField
            name="utilities"
            label="Utilities"
            value={formikProps.values.utilitiesPer}
            handlechange={formikProps.handlechange}
            handleblur={formikProps.handleblur}
          />
          <ExpenseField
            name="household"
            label="Household"
            value={formikProps.values.householdPer}
            handlechange={formikProps.handlechange}
            handleblur={formikProps.handleblur}
          />
          <ExpenseField
            name="tv"
            label="TV & Communications"
            value={formikProps.values.tvPer}
            handlechange={formikProps.handlechange}
            handleblur={formikProps.handleblur}
          />
          <ExpenseField
            name="other"
            label="Other"
            value={formikProps.values.otherPer}
            handlechange={formikProps.handlechange}
            handleblur={formikProps.handleblur}
          />

          {/* Diplay sum of all expenses */}
          <div>
            <br />
            <hr />
            <h3>Estimated Total Expenses</h3>
            <p>
              $
              <span className="bigDigits">
                {(formikProps.values.utilities || 0) +
                  (formikProps.values.household || 0) +
                  (formikProps.values.tv || 0) +
                  (formikProps.values.other || 0)}
              </span>{" "}
              /month
            </p>
          </div>
        </div>
        <div className="submitFooter">
          <button type="button" onClick={prevPage}>
            {"< Prev"}
          </button>
          <button type="submit">Submit</button>
        </div>
        </>
    );
  }
