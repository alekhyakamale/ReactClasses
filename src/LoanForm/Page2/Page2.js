import React, { Component } from "react";

import { MyInput, CustomRadioGroup, ExpenseField } from "../../Data/Data";

export default class Page1 extends Component {
  render() {
    return (
      <>
        <div className="div-color">
          <h2>Your Income</h2>
          <div>
            <MyInput
              type="number"
              label="What is you Salary?"
              span="$"
              name="salary"
            />
            <span>
              {" "}
              per{" "}
              <CustomRadioGroup
                type="radio"
                name="incomeper"
                value={this.props.formikProps.values.incomeper}
                options={["year", "month", "fortnight", "week"]}
              />{" "}
            </span>
            <input type="checkbox" name="aftertax"
             value={this.props.formikProps.values.afterTax} />
            <label for="aftertax">After Tax</label>
          </div>
          <CustomRadioGroup
            type="radio"
            label="Do you receive any government allowances or pensions?"
            name="allowances"
            value={this.props.formikProps.values.allowances}
            options={["Yes", "No"]}
          />
          <CustomRadioGroup
            type="radio"
            label="Do you receive any rental income?"
            name="rental"
            value={this.props.formikProps.values.rental}
            options={["Yes", "No"]}
          />
          <CustomRadioGroup
            type="radio"
            label="Do you receive any other additional income? "
            name="additional"
            value={this.props.formikProps.values.additional}
            options={["Yes", "No"]}
          />
        </div>
        <div className="div-color">
          <h2>Your Expenses</h2>
          <p>
            Please provide an accurate view of your share of living expenses by
            completing the following fields. (If you donâ€™t have the expense
            please enter $0). This will help us understand your circumstances so
            we can assess your application and we may validate the expenses
            stated.
          </p>
          <ExpenseField
            name="utilities"
            label="Utilities"
            selected_period={this.props.formikProps.values["utilitiesPer"]}
            handleChange={this.props.formikProps.handleChange}
            handleBlur={this.props.formikProps.handleBlur}
          />
          <ExpenseField
            name="household"
            label="Household"
            selected_period={this.props.formikProps.values["householdPer"]}
            handleChange={this.props.formikProps.handleChange}
            handleBlur={this.props.formikProps.handleBlur}
          />
          <ExpenseField
            name="tv"
            label="TV & Communications"
            selected_period={this.props.formikProps.values["tvPer"]}
            handleChange={this.props.formikProps.handleChange}
            handleBlur={this.props.formikProps.handleBlur}
          />
          <ExpenseField
            name="other"
            label="Other"
            selected_period={this.props.formikProps.values["otherPer"]}
            handleChange={this.props.formikProps.handleChange}
            handleBlur={this.props.formikProps.handleBlur}
          />

          {/* Diplay sum of all expenses */}
          <div className="row">
            <br />
            <hr />
            <h3>Estimated Total Expenses</h3>
            <p>
              $
              <span className="bigDigits">
                {(this.props.formikProps.values.utilities || 0) +
                  (this.props.formikProps.values.household || 0) +
                  (this.props.formikProps.values.tv || 0) +
                  (this.props.formikProps.values.other || 0)}
              </span>{" "}
              /month
            </p>
          </div>
        </div>
        <div className="submitFooter">
          <button type="button" onClick={this.props.prevPage}>
            {"< Prev"}
          </button>
          <button type="submit">Submit</button>
        </div>
      </>
    );
  }
}
