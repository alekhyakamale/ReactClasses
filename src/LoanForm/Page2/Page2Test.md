Test scenario: 
    Check input functionalities
    and submit the form

Test cases:
   1. Check response on entering Salary.
   2. Check response after selecting required options in the loan form. 
   3. Check response on entering expense fields

Test steps:
    1. Enter Salary that falls in range of 0-100000 only.
    2. Select one of the input options for all the input fields 
        within given options.
    3. Select an option from the drop down menu or leave it at default value.
    4. Enter values in expense fields that fall in range of 0-100000 only.
    5. Click on "Prev" in order to make changes in the previous page.
    6. Click on Submit after filling out all the required fields and selecting
        all the fitting options.

Expected result:
    The form page should not show any errors and the Submit button should work.