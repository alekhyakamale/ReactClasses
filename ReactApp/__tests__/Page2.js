import React from 'react';
import Page2 from '../src/LoanForm/Page2/Page2'
import {mount,  configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-extended';

configure({ adapter: new Adapter()})


it("render without crashing", () => {
    const wrapper = shallow(<Page2/>)
    expect(wrapper.getElements()).toMatchSnapshot();
})

