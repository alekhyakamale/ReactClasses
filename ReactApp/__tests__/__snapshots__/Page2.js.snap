// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render without crashing 1`] = `
Array [
  <React.Fragment>
    [Function]
  </React.Fragment>,
]
`;
